### ISSUE TITLE
Add problem detection for docker start failures

### ISSUE THREAD

**Issue by tstromberg** [kubernetes/minikube#4273]
See #4231 for an example.

**Comment by fenglixa** [comment]
@tstromberg I was able to duplicate this issue. But I am sorry, could you please clarify more about what behavior is expected for this ticket? Thanks very much!

BR
-Francis

**Comment by fenglixa** [comment]
Current behavior of starting docker is ahead of configuring, which mean this issue was already addressed? 
As following:
```bash
 ✘ ~/Documents/GitHub/minikube_fengli:$ sudo minikube start --registry-mirror registry.cn-hangzhou.aliyuncs.com/google_containers -p registry-mirror
😄  minikube v1.1.0 on darwin (amd64)
🔥  Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...

💣  Failed to enable container runtime: command failed: sudo systemctl start docker
stdout:
stderr: Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
: Process exited with status 1

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new
```

**Comment by afbjorklund** [comment]
I think the idea is to give a more helpful error message, when you use --registry-mirror without a scheme.

      --image-repository string           Alternative image repository to pull docker images from.
> This can be used when you have limited access to gcr.io. Set it to "auto" to let minikube decide one for you. For Chinese mainland users, you may use local gcr.io mirrors such as registry.cn-hangzhou.aliyuncs.com/google_containers

      --registry-mirror strings           Registry mirrors to pass to the Docker daemon
       --registry-mirror=<scheme>://<host>
> Prepend a registry mirror to be used for image pulls. May be specified multiple times.

Maybe even suggest "did you mean --image-repository", if it doesn't contain a :// and specify a scheme ?

**Comment by priyawadhwa** [comment]
Looks like a more helpful registry-mirror error message was included in #4284, so I'm going to go ahead and close this issue.
